plugin.php

```diff
@@ -106,6 +104,8 @@ function () {
 	$provider = new $selected_email_provider();
 
 	$newsletter_builder_email_provider = new $provider();
-	$newsletter_builder_email_provider->setup();
+	if ( $newsletter_builder_email_provider instanceof Email_Providers\Campaign_Monitor ) {
+		$newsletter_builder_email_provider->setup();
+	}
```

Comment:
Do we need to throw any kind of error or debug info if this check fails?

Reply:
That's a good idea. I just added an admin notice 👍